### PR TITLE
Meson library config

### PR DESCRIPTION
**Pull request by slci**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Core Types defines common classes and functionality used by multiple Functional
 
 ```sh
 mkdir build && cd build
-meson setup -Db_coverage=true --pkg-config-path .
```
Comment (bburda) [inline README.md:11]: Please change it also in `.travis.yml`

File: meson.build [hunk 2]
```diff
@@ -1,7 +1,13 @@
-project('ap-functional-cluster-template', 'cpp',
+project('ap-coretypes', 'cpp',
     version: '0.1',
     license: 'MIT',
-    default_options : ['cpp_std=c++2a']
+    default_options : [
+        'cpp_std=c++2a',
+        'warning_level=3',
+        'werror=true',
```
Comment (bburda) [inline meson.build:7]: @pz0 was adding compiler flags in https://github.com/UmlautSoftwareDevelopmentAccount/AP-Functional-Cluster-Template/pull/11 so maybe he can look at this?

**Comment by bburda** [comment]
This changes should be also added in some form to the https://github.com/UmlautSoftwareDevelopmentAccount/AP-Functional-Cluster-Template

File: meson.build (hunk 2 above)
Comment (slci) [reply meson.build:7]: Yes, absolutely. This provides a default warnings set that still can be extended. One just need to check what warnings Meson enables at level=3.

**Comment by slci** [comment]
@bburda This change solves #28 #32 but I don't think I can link Issues to the PR.

File: README.md (hunk 1 above)
Comment (slci) [reply README.md:11]: done

**Comment by bburda** [comment]
> @bburda This change solves #28 #32 but I don't think I can link Issues to the PR.

Linked.

**Pull request merged**